[AIRFLOW-6786] Added Kafka components

dferguson992 opened the pull request:
Dear Airflow Maintainers,

Please accept the following PR that

Add the KafkaProducerHook.
Add the KafkaConsumerHook.
Add the KafkaSensor which listens to messages with a specific topic.
Related Issue:
#1311

Issue link: AIRFLOW-6786

Make sure to mark the boxes below before creating PR: [x]

 Description above provides context of the change
 Commit message/PR title starts with [AIRFLOW-NNNN]. AIRFLOW-NNNN = JIRA ID*
 Unit tests coverage for changes (not needed for documentation changes)
 Commits follow "How to write a good git commit message"
 Relevant documentation is updated including usage instructions.
 I will engage committers as explained in Contribution Workflow Example.
For document-only changes commit message can start with [AIRFLOW-XXXX].
Reminder to contributors:

You must add an Apache License header to all new files
Please squash your commits when possible and follow the 7 rules of good Git commits
I am new to the community, I am not sure the files are at the right place or missing anything.

The sensor could be used as the first node of a dag where the second node can be a TriggerDagRunOperator. The messages are polled in a batch and the dag runs are dynamically generated.

Thanks!

Note, as per denied PR #1415, it is important to mention these integrations are not suitable for low-latency/high-throughput/streaming. For reference, #1415 (comment).

Co-authored-by: Dan Ferguson dferguson992@gmail.com
Co-authored-by: YuanfΞi Zhu

Commit:
Added Kafka components

Addressing documentation errors

Addressing documentation errors

Adding kafka installation requirement to setup

Update docs/operators-and-hooks-ref.rst

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

Added dag start date back to example dag

???

????

Adds 'cncf.kubernetes' package back to backport provider packages. (#10659)

Display conf as a JSON in the DagRun list view (#10644)

Co-authored-by: Marco Aguiar <marco@DESKTOP-8IVSCHM.localdomain>

Unify error messages and complete type field in response (#10333)

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

Add howto doc for Salesforce connection (#10482)

Add example on airflow users create --help (#10662)

Added some documentation requirements

Addressing documentation errors

Addressing documentation errors

Adding kafka installation requirement to setup

Update docs/operators-and-hooks-ref.rst

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

Added dag start date back to example dag

???

????

Adds 'cncf.kubernetes' package back to backport provider packages. (#10659)

Display conf as a JSON in the DagRun list view (#10644)

Co-authored-by: Marco Aguiar <marco@DESKTOP-8IVSCHM.localdomain>

Unify error messages and complete type field in response (#10333)

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

Add howto doc for Salesforce connection (#10482)

Add example on airflow users create --help (#10662)

Added some documentation requirements

Commit:
Added Kafka components

potiuk committed:
Adds 'cncf.kubernetes' package back to backport provider packages. (#10659)

marcoaaguiar committed:
Display conf as a JSON in the DagRun list view (#10644)

Co-authored-by: Marco Aguiar <marco@DESKTOP-8IVSCHM.localdomain>

ephraimbuddy committed:
Unify error messages and complete type field in response (#10333)

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

Commit:
Addressing documentation errors

mik-laj commented on docs/operators-and-hooks-ref.rst:
```diff
@@ -176,6 +176,14 @@ Foundation.
        :mod:`airflow.providers.apache.hive.sensors.hive_partition`,
        :mod:`airflow.providers.apache.hive.sensors.metastore_partition`
 
+  * - `Apache Kafka <https://kafka.apache.org/>`__
```
```suggestion
   * - `Apache Kafka <https://kafka.apache.org/>`__
```

Commit:
Addressing documentation errors

ryw reviewed (commented):
I'm generally in favor of Airflow having Kafka integration, so +1 on this conceptually

eladkal committed:
Add howto doc for Salesforce connection (#10482)

mik-laj committed:
Add example on airflow users create --help (#10662)

potiuk committed:
Remove requirements from the project. (#10668)

The requirements are not needed any more. We replaced them
with a new, better "constraints" mechanism where constraints
are stored in a separate, orphaned branches in the repository
and they are automatically maintained by the CI process.

See more about our dependency management process here:

https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pinned-constraint-files

Commit:
Added some documentation requirements

kanga333 committed:
Add placement_strategy option (#9444)

michalslowikowski00 committed:
Add Dataprep operators (#10304)

Add DataprepGetJobGroupOperator and DataprepRunJobGroupOperator
for Dataprep service.

Co-authored-by: Tomek Urbaszek <tomasz.urbaszek@polidea.com>

potiuk committed:
Add packages to function names in bash (#10670)

Inspired by the Google Shell Guide where they mentioned
separating package names with :: I realized that this was
one of the missing pieces in the bash scripts of ours.

While we already had packages (in libraries folders)
it's been difficult to realise which function is where.

With introducing packages - equal to the library file name
we are *almost* at a level of a structured language - and
it's easier to find the functions if you are looking for them.

Way easier in fact.

Part of #10576

potiuk committed:
Add packages to function names in bash (#10670)

Inspired by the Google Shell Guide where they mentioned
separating package names with :: I realized that this was
one of the missing pieces in the bash scripts of ours.

While we already had packages (in libraries folders)
it's been difficult to realise which function is where.

With introducing packages - equal to the library file name
we are *almost* at a level of a structured language - and
it's easier to find the functions if you are looking for them.

Way easier in fact.

Part of #10576

eyalzek committed:
Add `log_id` field to log lines on ES handler (#10411)

* Add `log_id` field to log lines on ES handler

* Add `offset` field to log lines on ES handler

it will be set to the epoch timestamp in nanoseconds (this will just be
used for ordering log lines when displayed in the webserver UI).

* Update UPDATING.md

With information regarding log_id and offset fields in JSON log lines written to stdout

Commit:
Removed the requirements files

Commit:
Fixing doc issue

Commit:
Files modified by black

ryw committed:
Fix format of install commands (#10676)

ryw committed:
Improve getting started section (#10680)

ryw committed:
Remove airflow-pr tool  (#10675)

* Remove airflow-pr tool

* Add PyGithub back in

* Remove gitpython

NickBenthem committed:
Update INTHEWILD.md (#10683)

kaxil committed:
Remove redundant section from dev/README.md toc (#10689)

mik-laj committed:
[AIRFLOW-XXX] Add task execution process on Celery Execution diagram (#6961)

yene committed:
docs: They added support for celltags to Jupyter Lab (#9141)

mik-laj committed:
Unify command names in CLI (#10669)

* Unify command names in CLI

dimberman committed:
Add on_kill support for the KubernetesPodOperator (#10666)

This PR ensures that when a user kills a KubernetesPodOperator task
in the airflow UI, that the associated pod is also killed using the
on_kill method.

kaxil committed:
Revert recent breeze changes (#10651 & #10670) (#10694)

* Revert "Add packages to function names in bash (#10670)"

This reverts commit cc551ba793344800d2d396c13d7fd0c8eed97352.

* Revert "Implement Google Shell Conventions for breeze script … (#10651)"

This reverts commit 46c8d6714c981746cc114b8b1af5cb27aa0018e2.

kaxil committed:
Fix failing black test (#10697)

* Fix failing black test

olchas committed:
Fix missing dash in flag for statsd container (#10691)

Co-authored-by: Kamil Olszewski <kamil.olszewski@polidea.com>

kaxil committed:
Chart: Flower deployment should use Flower image (#10701)

Co-authored-by: Steven Miller <sjmiller609@gmail.com>

potiuk committed:
Implement Google Shell Conventions for breeze script (#10695)

* Implement Google Shell Conventions for breeze script … (#10651)

Part of #10576

First (and the biggest of the series of commits to introduce
Google Shell Conventions in our bash scripts.

This is about the biggest and the most complex breeze script
so it is rather huge but it is difficult to split it into
smaller pieces.

The rules implemented (from the conventions):

 * constants and exported variables are CAPITALIZED, where
   local/temporary variables are lowercase

 * following the shell guide, once all the variables are set to their
   final values (either from exported variables, calculation or --switches
   ) I have a single function that makes all the variables read-only. That
   helped to clean-up a lot of places where same functions was called
   several times, or where variables were defined in a few places. Now the
   behavior should be rather consistent and we should easily catch some
   duplications

 * function headers (following the guide) explaining arguments,
   variables expected, variables modified in the functions used.

 * setting the variables as read-only also helped to clean-up the "ifs"
   where we often had ":=}" in variables and != "" or == "". Those are
   replaced with `=}` and tests are replaced with `-n` and `-z` - also
   following the shell guide (readonly helped to detect and clean all
   such cases). This also should be much more robust in the future.

 * reorganized initialization of those constants and variables - simplified
   a few places where initialization was overlapping. It should be much more
   straightforward and clean now

 * a number of internal function breeze variables are "local" - this is
   helpful in accidental variables overwriting and keeping stuff localized

 * trap_add function is separated out to help in cases where we had
   several traps handling the same signals.

(cherry picked from commit 46c8d6714c981746cc114b8b1af5cb27aa0018e2)
(cherry picked from commit c822fd7b4bf2a9c5a9bb3c6e783cbea9dac37246)

* fixup! Implement Google Shell Conventions for breeze script … (#10651)

potiuk committed:
Add packages to function names in bash (#10670) (#10696)

Inspired by the Google Shell Guide where they mentioned
separating package names with :: I realized that this was
one of the missing pieces in the bash scripts of ours.

While we already had packages (in libraries folders)
it's been difficult to realise which function is where.

With introducing packages - equal to the library file name
we are *almost* at a level of a structured language - and
it's easier to find the functions if you are looking for them.

Way easier in fact.

Part of #10576

(cherry picked from commit cc551ba793344800d2d396c13d7fd0c8eed97352)
(cherry picked from commit 2bba276f0f06a5981bdd7e4f0e7e5ca2fe84f063)

lopesdiego12 committed:
Update INTHEWILD.md (#10703)

kaxil committed:
Change the name of Static Check without pylint (#10690)

lordozb committed:
Add jupytercmd and fix task failure when notify set as true in qubole operator (#10599)

Add jupytercmd in Qubole Operator which fires a JupyterNotebookCommand to the jupyter notebooks running on user's QDS account. Along with this, we have fixed a minor bug that caused the tasks to fail with --notify is set in Qubole Operator.

Co-authored-by: Aaditya Sharma <asharma@qubole.com>

turbaszek committed:
Make Cloud Build system tests setup runnable (#10692)

This change fixes error: open(quickstart.sh): Permission denied
that was rised during git add.

kaxil committed:
[AIRFLOW-5948] Replace SimpleDag with SerializedDag (#7694)

mik-laj committed:
Update DAG Serialization docs (#10711)

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

jpmarques66 committed:
Migrate speccy to spectral in OpenAPI linting. (#10351)

ashb committed:
Don't commit when explicitly passed a session to TI.set_state (#10710)

The `@provide_session` wrapper will already commit the transaction when
returned, unless an explicit session is passed in -- removing this
parameter changes the behaviour to be:

- If session explicitly passed in: don't commit (caller's
  responsibility)
- If no session passed in, `@provide_session` will commit for us already.

ashb committed:
Make test_trigger_rule_dep tests re-runnable (#10712)

If we run this test
(TestTriggerRuleDep::test_get_states_count_upstream_ti specifically)
more than once without clearing the DB in between it would fail due to a
unique constraint violation.

ashb committed:
Ensure we heartbeat the DagFileProcessorManager regularly. (#10706)

It could have wedged, (but the process still be alive) and we would
never notice.

In this I use `time.monotonic` rather than a `datetime` object for two
reasons:

1. We don't need the expense of a "full" date time object since all we
    care about is the second diff between two points in time.
2. It is "more correct" as `datetime.now()` would be inaccurate if the
   system clock changes (NTP etc.)

msumit committed:
Add Stacktrace when DagFileProcessorManager gets killed (#10681)

chrismclennon committed:
Add Indeed to INTHEWILD.md (#10716)

mik-laj committed:
Unify command names in CLI (#10720)

* Unify command names in CLI

* fixup! Unify command names in CLI

dimberman committed:
Add generate_yaml command to easily test KubernetesExecutor before deploying pods (#10677)

* Add generate_template command for kubernetes_executor

* move import

* fix test failure

* Address @mik-laj comments

* Address @mik-laj comments

* Use current dir

* add docs

* fix test

kaxil committed:
Relax requirement to allow latest version of flask-login (#9990)

The current latest available version is 0.5.0 (https://pypi.org/project/Flask-Login/0.5.0/)

Version 0.5.0 drops support for Python 2.6, 3.3, 3.4 and we don't use those versions in Airflow Master

dimberman committed:
Make grace_period_seconds option on K8sPodOperator (#10727)

* Make grace_period_seconds option on K8sPodOperator

This PR allows users to choose whether they want to gracefully kill
pods when they delete tasks in the UI or if they would like to
immediately kill them.

* Update airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

potiuk committed:
Switches to better BATS asserts (#10718)

BATS has additional libraries of asserts that are much more
straightforward and nicer to write tests for bash scripts

There is no dockerfile from BATS that contains those, so we
had to build our own (but it follows the same structure
as #9652 - where we keep our dev docker image
sources inside our repository and the generated docker images
in "apache/airflow:<tool>-CALVER-TOOLVER format.

We have more BATS unit test to add - following #10576
and this change will be of great help.

antoniocali committed:
Add masterConfig parameter to MLEngineStartTrainingJobOperator (#10578)

Co-authored-by: antonio-davide-cali <antonio.davide.cali@ikea.com>

Commit:
Add securitySchemes in openapi spec (#10652)

openapi-generator relies on this component to generate auth code in
some of the clients.

kaxil committed:
Fix docs for generate-dag-yaml cli command (#10735)

kaxil committed:
Enable more checks for pydocstyle (#10741)

Enable D106, D207 and D208

D106	Missing docstring in public nested class
D207	Docstring is under-indented
D208	Docstring is over-indented

ryanyuan committed:
Update node installation cmd (#10744)

kaxil committed:
Improve test coverage for test_common_schema.py (#10740)

Adds test that an error is raised with specific message when unkown object type is passed

kaxil committed:
Fix typo in test_dag_run_schema.py (#10739)

kaxil committed:
Improve test coverage for ConfObject in dag_run_schema (#10738)

Adds test to verify that string can be passed to conf and ConfObject._deserialize works.

kaxil committed:
Add black to STATIC_CODE_CHECKS.rst (#10737)

Add black to the table for Static Code Checkers

kaxil committed:
Remove duplicate entries from .mailmap (#10736)

rootcss committed:
Add docs for how airflow manages packages and imports (#10303)

varundhussa committed:
Asynchronous execution of Dataproc jobs with a Sensor (#10673)

kushsharma committed:
clean-logs script for Dockerfile: trim logs before sleep (#10685)

If the pod restarts before the sleep time is over, the trim command will not run. I think it's better if we reorder the commands to execute the delete and then go to sleep. At the moment sleep is every 15 mins but people will just increase the EVERY line if they want longer sleep time and can encounter this bug.

potiuk committed:
Make scripts/ci/openapi Google Shell Guide compatible (#10747)

Part of #10576

potiuk committed:
Switch to downloaded pgbouncer_exporter (#10759)

Fixes #10753

ashb committed:
Fix `breeze flags` command. (#10766)

This was missed in #10670

mik-laj committed:
Move role guide to access control (#10755)

potiuk committed:
Check all dockerfiles with hadolint (#10754)

The hadolint check only checked the "main dir" Dockerfile
but we have more of them now. All of them are now checked.

The following problems are fixed:

 * DL3000 Use absolute WORKDIR
 * DL4000 MAINTAINER is deprecated
 * DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it.
 * SC2046 Quote this to prevent word splitting.

The followiing problems are ignored:

 * DL3018 Pin versions in apk add. Instead of `apk add <package>` use `apk add
   <package>=<version>`

jedcunningham committed:
Add permission "extra_links" for Viewer role and above (#10719)

This change adds 'can extra links on Airflow' to the Viewer role and above. Currently, only Admins can see extra links by default.

potiuk committed:
The verbose functions will not exit immediately if not asked to (#10731)

The docker(), helm(), kubectl() functions replace the real tools
to get verbose behaviour (we can print the exact command being
executed for those. But when 'set +e' was set before the command
was called - indicating that error in those functions should be
ignored - this did not happen. The functions set 'set -e' just
before returning the non-zero value, effectively exiting the
script right after. This caused first time experience to be not
good.

The fix also fixes behaviour of stdout and stderr for those
functions - previously they were joined to be able to be
printed to OUTPUT_FILE but this lost the stderr/stdout
distinction. Now both stdout and stderr are printed to the
output file but they are also redirected to stdout/stderr
respectively, so that 2>/dev/null works as expected.

While fixing it, it turned out that one of the remove_images
methods was not used any more - merged it with the breeze one.

potiuk committed:
Make ci/scripts/pre-commit Google Shell Guide compatible (#10748)

Part of #10576

mik-laj committed:
Simplify load connection in LocalFilesystemBackend (#10638)

potiuk committed:
The scripts to run tests properly initialises constants (#10769)

The constants were initialised after the readonly status was set
for the constants in the test script.

This was mainly about default values for those consttants (but this
has already been handled by the _script_init.sh but more importantly
the INTEGRATIONS were not properly initialized that cause skipping of
some integration tests.

potiuk committed:
Make static checks Google Shell Guide compatible (#10750)

Part of #10576

turbaszek committed:
Refactor DataprocCreateCluster operator to use simpler interface (#10403)

DataprocCreateCluster requires now:
- cluster config
- cluster name
- project id

In this way users don't have to pass project_id two times 
(in cluster definition and as parameter). The cluster object 
is built in create_cluster hook method

MateuszKukiela committed:
[AIRFLOW-10672] Refactor BigQueryToGCSOperator to use new method (#10773)

Makes BigQueryToGCSOperator to use BigQueryHook.insert_job method

Committer: Mateusz Kukieła <mateuszkukiela@gmail.com>

potiuk committed:
Fixes pre-commit failing on build step (#10785)

When rebuildig the image during commit, kill command failed to
find the spinner job to kill (this is just preventive measure)
and failed the rebuild step in pre-commit.

This is now fixed.

Commit:
fix task instance modal open performance issue (#10764)

This reduces the task instance modal open wait time for large DAGs with
extra links from minutes to 6 ms, an order of 10000 times speed up.

potiuk committed:
Removes stable tests from quarantine (#10768)

We've observed the tests for last couple of weeks and it seems
most of the tests marked with "quarantine" marker are succeeding
in a stable way (https://github.com/apache/airflow/issues/10118)
The removed tests have success ratio of > 95% (20 runs without
problems) and this has been verified a week ago as well,
so it seems they are rather stable.

There are literally few that are either failing or causing
the Quarantined builds to hang. I manually reviewed the
master tests that failed for last few weeks and added the
tests that are causing the build to hang.

Seems that stability has improved - which might be casued
by some temporary problems when we marked the quarantined builds
or too "generous" way of marking test as quarantined, or
maybe improvement comes from the #10368 as the docker engine
and machines used to run the builds in GitHub experience far
less load (image builds are executed in separate builds) so
it might be that resource usage is decreased. Another reason
might be Github Actions stability improvements.

Or simply those tests are more stable when run isolation.

We might still add failing tests back as soon we see them behave
in a flaky way.

The remaining quarantined tests that need to be fixed:
 * test_local_run (often hangs the build)
 * test_retry_handling_job
 * test_clear_multiple_external_task_marker
 * test_should_force_kill_process
 * test_change_state_for_tis_without_dagrun
 * test_cli_webserver_background

We also move some of those tests to "heisentests" category
Those testst run fine in isolation but fail
the builds when run with all other tests:
 * TestImpersonation tests

We might find that those heisentest can be fixed but for
now we are going to run them in isolation.

Also - since those quarantined tests are failing more often
the "num runs" to track for those has been decreased to 10
to keep track of 10 last runs only.

dmitrikuksik committed:
Deprecate using global as the default region in Google Dataproc operators and hooks (#10772)

The region parameter is required for some of Google Dataproc operators
and it should be provided by users to avoid creating data-intensive 
tasks in any default location.

jmcarp committed:
Always return a list from S3Hook list methods (#10774)

potiuk committed:
Move dev docker images to airflow registry (#9652)

Part of #9401

ephraimbuddy committed:
Add unit test for AzureCosmosDocumentSensor (#10765)

ephraimbuddy committed:
Extract missing gcs_to_local example DAG from gcs example (#10767)

Co-authored-by: Kamil Breguła <kamil.bregula@polidea.com>

mik-laj committed:
All files in providers package heve unit tests (#10799)

potiuk committed:
Check that all pre-commits are synchronized code<>docs (#10789)

Until pre-commit implements export of all configured
checks, we need to maintain the list manually updated.

We check both - pre-commit list in breeze-complete and
descriptions in STATIC_CODE_CHECKS.rst

ryw committed:
Add section for official source code (#10678)

* Add section for official source code

* Update README.md

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

* Update README.md

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

* Update README.md

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

* Fix typos

* Fix capitalization

* Update README.md

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

* Update README.md

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

* Format fixup

* Remove official docker images section

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

ryw committed:
Refactor official source section to use bullets (#10801)

* Refactor section to use bullets

* Change structure to group the binary options

potiuk committed:
Make script/ci/images Google Shell Guide compatible (#10745)

Part of #10576

potiuk committed:
Make script/ci/kubernetes Google Shell Guide Compatible (#10746)

Part of #10576

potiuk committed:
Fixed wrong "-e" on md5 file status check (#10803)

The "-e" flag was not reset properly in the md5 status check
which could lead in some cases to removing output of flake check.

potiuk committed:
Make ci/backport_packages Google Shell guide compliant (#10733)

potiuk committed:
Fix integration tests being accidentally excluded (#10807)

The change from #10769 accidentally switched Integration tests
into far-longer run unit tests (we effectively run the tests
twice and did not run integration tests.

This fixes the problem by removing readonly status from
INTEGRATIONS and only setting it after the integrations are
set.

potiuk committed:
Move Impersonation test back to quarantine (#10809)

Seems that TestImpersonation is not stable even in isolation
Moving it back to quarantine for now.

ashb committed:
Don't include kubernetes_tests/ and backport_packages/ in our wheel (#10805)

Since 1.10.x we have added some new top-level folders that were getting
included in the build wheel!

mik-laj committed:
Add task logging handler to airflow info command (#10771)

YingboWang committed:
[AIRFLOW-3964][AIP-17] Consolidate and de-dup sensor tasks using Smart Sensor (#5499)

Co-authored-by: Yingbo Wang <yingbo.wang@airbnb.com>

dimberman committed:
Add pod_override setting for KubernetesExecutor (#10756)

* Add podOverride setting for KubernetesExecutor

Users of the KubernetesExecutor will now have a "podOverride"
option in the executor_config. This option will allow users to
modify the pod launched by the KubernetesExecutor using a
`kubernetes.client.models.V1Pod` class. This is the first step
in deprecating the tradition executor_config.

* Fix k8s tests

* fix docs

kaxil committed:
Update log exception to reflect rename of execute_helper (#10819)

`SchedulerJob.execute_helper` was renamed to `SchedulerJob._run_scheduler_loop`

kaxil committed:
Upgrade black to 20.8b1 (#10818)

turbaszek committed:
Add connection caching to KubernetesHook (#10447)

* Add connection caching to KubernetesHook

* Fix recursion and remove redundant docstrings

potiuk committed:
Make airflow testing Google Shell Guide compatible (#10813)

Part of #10576

potiuk committed:
Make scripts/ci/tools Google Shell Guide Compatible (#10811)

Part of #10576

atsalolikhin-spokeo committed:
Add documentation for preparing database for Airflow (#10413)

potiuk committed:
Make dockerfiles Google Shell Guide Compliant (#10734)

Part of #10576

ashb committed:
Fix doc errors introduced in #10413 (#10832)

Co-authored-by: Kamil Breguła <mik-laj@users.noreply.github.com>

ashb committed:
Fix typos in the stackdriver howto document (#10834)

Someone was overzealous in building the initial ignore list -- "wil" is
not a word :)

And the spell checker does not catch repeated words.

mik-laj committed:
Fix doc errors introduced in #10413 (#10833)

kaxil committed:
Fix grammar in UPDATING.md (#10841)

`changes` -> `changed`

dimberman committed:
Remove k8s dependency from serialization (#10831)

Ensures that airflow can run without downloading the
Kubernetes python client

kaxil committed:
Fix typo in docs/howto/operator/amazon/aws/emr.rst (#10844)

`Documetation` -> `Documentation`

kaxil committed:
Add missing assert call in test_dbapi_hook.py (#10842)

`assert` call was missing so the statement didn't test or wouldn't fail if condition isn't true

kaxil committed:
Fix and remove some more typos from spelling_wordlist.txt (#10845)

kaxil committed:
Add test for Health Endpoint when there is an exception (#10846)

abhilash1in committed:
Add USC Graduate School to INTHEWILD.md (#10843)

potiuk committed:
Move parse_once to quarantine (#10857)

ryw committed:
Rename "Beyond the Horizon" section and refactor content (#10802)

* Rename "Beyond the Horizon" section and refactor content

* Rework copy

ashb committed:
Detect orphaned task instances by SchedulerJob id and heartbeat (#10729)

Once HA mode for scheduler lands, we can no longer reset orphaned
task by looking at the tasks in (the memory of) the current executor.

This changes it to keep track of which (Scheduler)Job queued/scheduled a
TaskInstance (the new "queued_by_job_id" column stored against
TaskInstance table), and then we can use the existing heartbeat
mechanism for jobs to notice when a TI should be reset.

As part of this the existing implementation of
`reset_state_for_orphaned_tasks` has been moved out of BaseJob in to
BackfillJob -- as only this and SchedulerJob had these methods, and the
SchedulerJob version now operates differently

ashb committed:
Fix and unquarantine TestDagFileProcessorAgent.test_parse_once (#10862)

The SmartSensor PR introduces slightly different behaviour on
list_py_files happens when given a file path directly.

Prior to that PR, if given a file path it would not include examples.

After that PR was merged, it would return that path and the example dags
(assuming they were enabled.)

ryw committed:
Refactor content to a markdown table (#10863)

Commit:
Add on_kill method to DataprocSubmitJobOperator (#10847)

ashb committed:
Add new lint check to now allow realtive imports (#10825)

Relative and absolute imports are functionally equivalent, the only
pratical difference is that relative is shorter.

But it is also less obvious what exactly is imported, and harder to find
such imports with simple tools (such as grep).

Thus we have decided that Airflow house style is to use absolute imports
only

millertracy committed:
Added Plexus as an Airflow provider (#10591)

RosterIn commented on airflow/providers/google/cloud/example_dags/example_dataproc.py:
```diff
@@ -65,10 +65,16 @@
 # Update options
 # [START how_to_cloud_dataproc_updatemask_cluster_operator]
 CLUSTER_UPDATE = {
-    "config": {"worker_config": {"num_instances": 3}, "secondary_worker_config": {"num_instances": 3}}
```
How is adding kafka integration related to modifying google dag examples?

dferguson992 replied:
idk how those even got in.  Whenever i can work on this, i rebase from master and a whole bunch of changes get added..  i think I'm rebasing wrong but i don't understand how.  `git rebase -i <my_branch> master`?

I never have enough time to really isolate the issue with this PR, especially since everytime i try toupdate my local branch i incur other commits that I didn't write.  not sure what's wrong with this at all.

RosterIn replied:
There is a really good guide at the contributing section:
https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#id9
On slack there is a channel airflow-how-to-pr where you can ask for help if you get into trouble.

dferguson992 replied:
i've read through this guide and i'm still not sure what's happening here.  I've joined the slack channel and raised my concerns there as well
